### PR TITLE
Fixture/bit extract

### PR DESCRIPTION
**Pull request by agnesLeroy**
### Resolves:

closes https://github.com/zama-ai/concrete-core-internal/issues/411
closes https://github.com/zama-ai/concrete-core-internal/issues/428

### Checklist 

(Use '[x]' to check the checkboxes, or submit the PR and then click the checkboxes)

* [x] Tests for the changes have been added (for bug fixes / features)
* [x] Docs have been added / updated (for bug fixes / features)
* [x] The PR description links to the related issue (to link an issue, use '#XXX'.)
* [x] The tests on AWS have been launched and are successful (comment with @slab-ci cpu_test and/or @slab-ci gpu_test to trigger the tests)
* [x] The draft release description has been updated
* [x] Check for breaking changes (including serialization changes) and add them to commit message following the conventional commit [specification][conventional-breaking]

[conventional-breaking]: https://www.conventionalcommits.org/en/v1.0.0/#commit-message-with-description-and-breaking-change-footer

### CODE REVIEW

**Comment by IceTDrinker** [comment]
@agnesLeroy I can explain the main changes I made, also I put a unit testing part in there as we will be going towards that more and more from what I understand

**Comment by IceTDrinker** [comment]
@slab-ci cpu_test

**Comment by IceTDrinker** [comment]
we may want to ask for actual parameters/use our tools to generate them for some security level potentially

**Comment by IceTDrinker** [comment]
@agnesLeroy I can't request a review as this is your PR initially, I let you add comment anyways so that we can work out the kinks and then we'll see to merge

File: concrete-core-fixture/src/fixture/lwe_ciphertext_discarding_bit_extraction.rs [hunk 1]
```diff
@@ -0,0 +1,532 @@
...
+
+        let recovered_extracted_bits: Vec<Precision::Raw> = actual
+            .iter()
+            .map(|&decrypted| {
+                decomposer.closest_representable(decrypted) >> (Precision::Raw::BITS - 1)
+            })
+            .collect();
+        let original_raw_bits: Vec<Precision::Raw> = means
+            .iter()
+            .map(|&raw_bit| raw_bit >> (Precision::Raw::BITS - 1))
```
Comment (agnesLeroy) [inline concrete-core-fixture/src/fixture/lwe_ciphertext_discarding_bit_extraction.rs:407]: Why is it necessary to make this shift on the original raw bits & the recovered ones?

File: concrete-core-fixture/src/fixture/lwe_ciphertext_discarding_bit_extraction.rs [hunk 2]
```diff
@@ -0,0 +1,532 @@
...
+        }
+        // Checks all results are true, returning the boolean itself
+        extracted_bits_are_the_same && result_per_bit.iter().all(|&result_ok| result_ok)
+    }
+}
+
+// FIXME:
+// The current NPE does not use the key distribution markers of concrete-core. This function makes
+// the mapping. This function should be removed as soon as the npe uses the types of concrete-core.
+#[allow(clippy::too_many_arguments)]
```
Comment (agnesLeroy) [inline concrete-core-fixture/src/fixture/lwe_ciphertext_discarding_bit_extraction.rs:446]: These formulas are not used in the end, or I missed that?

**Review by agnesLeroy — COMMENTED** [review]
Hello! Thanks a lot for this :) I just have 2 questions you'll see.

File: concrete-core-fixture/src/fixture/lwe_ciphertext_discarding_bit_extraction.rs (hunk 1 above)
Comment (IceTDrinker) [reply concrete-core-fixture/src/fixture/lwe_ciphertext_discarding_bit_extraction.rs:407]: Strictly it's not, but for assert/debug print it makes it easier to see 0/1 or some weird very big values

File: concrete-core-fixture/src/fixture/lwe_ciphertext_discarding_bit_extraction.rs (hunk 2 above)
Comment (IceTDrinker) [reply concrete-core-fixture/src/fixture/lwe_ciphertext_discarding_bit_extraction.rs:446]: In criteria line 353

File: concrete-core-fixture/src/fixture/lwe_ciphertext_discarding_bit_extraction.rs (hunk 1 above)
Comment (agnesLeroy) [reply concrete-core-fixture/src/fixture/lwe_ciphertext_discarding_bit_extraction.rs:407]: Ok thanks !

File: concrete-core-fixture/src/fixture/lwe_ciphertext_discarding_bit_extraction.rs (hunk 2 above)
Comment (agnesLeroy) [reply concrete-core-fixture/src/fixture/lwe_ciphertext_discarding_bit_extraction.rs:446]: Tanks :)

**Pull request merged**